contracts/MuonNodeManager.sol

```diff
@@ -0,0 +1,346 @@
+     * @param _lastEditTime The time of the last edit.
+     * @param _from The starting node ID.
+     * @param _to The ending node ID.
+     * @return nodesList An array of edited nodes.
+     */
+    function getEditedNodes(
+        uint256 _lastEditTime,
+        uint64 _from,
+        uint64 _to
+    ) public view returns (Node[] memory nodesList) {
+        _from = _from > 0 ? _from : 1;
+        _to = _to <= lastNodeId ? _to : lastNodeId;
+        require(_from <= _to, "Invalid range of node IDs.");
+
+        nodesList = new Node[](100);
+        uint64 n = 0;
+        for (uint64 i = _from; i <= _to && n < 100; i++) {
+            Node memory node = nodes[i];
+
+            if (node.lastEditTime > _lastEditTime) {
+                nodesList[n] = node;
+                nodesList[n].roles = getNodeRoles(i);
+                n++;
+            }
+        }
+
+        // Resize the array to remove any unused elements
+        assembly {
+            mstore(nodesList, n)
+        }
+    }
+
+    /**
+     * @dev Returns the information of a node associated with the provided node address.
+     * @param _addr The node address.
+     * @return node The node information.
+     */
+    function nodeAddressInfo(address _addr)
+        public
+        view
+        returns (Node memory node)
+    {
+        node = nodes[nodeAddressIds[_addr]];
+    }
+
+    /**
+     * @dev Returns the information of a node associated with the provided staker address.
+     * @param _addr The staker address.
+     * @return node The node information.
+     */
+    function stakerAddressInfo(address _addr)
+        public
+        view
+        override
+        returns (Node memory node)
+    {
+        node = nodes[stakerAddressIds[_addr]];
+    }
+
+    /**
+     * @dev Sets the tier of a node.
+     * Only callable by the DAO_ROLE.
+     * @param nodeId The ID of the node.
+     * @param tier The tier to set.
+     */
+    function setTier(uint64 nodeId, uint8 tier)
+        public
+        onlyRole(DAO_ROLE)
+        updateState
+        updateNodeState(nodeId)
+    {
+        nodes[nodeId].tier = tier;
+        emit TierSet(nodeId, tier);
+    }
+
+    /**
+     * @dev Sets a configuration value.
+     * Only callable by the DAO_ROLE.
+     * @param key The key of the configuration value.
+     * @param val The value to be set.
+     */
+    function setConfig(string memory key, string memory val)
+        public
+        onlyRole(DAO_ROLE)
+    {
+        configs[key] = val;
+        emit ConfigSet(key, val);
+    }
+
+    /**
+     * @dev Adds a new role.
+     * Only callable by the DAO_ROLE.
+     * @param role The role to be added.
+     */
+    function addNodeRole(bytes32 role) public onlyRole(DAO_ROLE) {
+        require(roleIds[role] == 0, "This role has already been added.");
+
+        lastRoleId++;
+        roleIds[role] = lastRoleId;
+        emit NodeRoleAdded(role, lastRoleId);
+    }
+
```

Comment:
Let's add a view function `getInfo(string configs[])` to be able to get all of contract info in one request.

lastUpdateTime
lastNodeId
lastRoleId
config values for the given configs

should be returned.